### PR TITLE
document feedback function usage

### PR DESCRIPTION
**Pull request by rshih32**
* Makes API docs actual APIs
* Added Huggingface/OpenAI/GroundTruth/Groundedness API code Snippets
* Moved the text descriptions of feedback functions to Feedback Function Guide section
* Link usage back and forth between the App and the Function APIs (so there is a full loop on usage in docs alone)

## Feedback Function APIs enabled
Screen Shot 2023-09-05 at 4 06 02 PM

## Feedback Function API before (No actual APIs)
Screen Shot 2023-09-05 at 4 05 52 PM

## AzureOpenAI Instantiation
Screen Shot 2023-09-05 at 4 07 17 PM

## Feedback Container Instantiation (Ground Truth)
Screen Shot 2023-09-05 at 4 07 37 PM

## Ground Truth Usage Docs
Screen Shot 2023-09-05 at 4 07 52 PM

## Feedback Functions link back to App Usage
Screen Shot 2023-09-05 at 4 06 14 PM

## TruChain/LLama/Basic/Custom Top Level Usage Docs (Usage and feedback function linkback)
Screen Shot 2023-09-05 at 4 06 38 PM

## TruChain/LLama/Basic/Custom Old Docs (No useful info)
Screen Shot 2023-09-05 at 4 06 53 PM

Todo:
- [ ] Make Selectors easier to find and describe

### CODE REVIEW

File: trulens_eval/examples/frameworks/custom/custom_example.ipynb [hunk 1]
```diff
@@ -85,10 +85,10 @@
    "metadata": {},
```
Comment (joshreini1) [inline trulens_eval/examples/frameworks/custom/custom_example.ipynb:85]: <p>Can you explain a bit more why this custom class example is useful?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/truera/trulens/pull/425/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (joshreini1) [inline trulens_eval/examples/frameworks/custom/custom_example.ipynb:85]: <p>Please remove dev cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/truera/trulens/pull/425/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (rshih32) [reply trulens_eval/examples/frameworks/custom/custom_example.ipynb]: accidentally changed from full repo string replace

Comment (rshih32) [reply trulens_eval/examples/frameworks/custom/custom_example.ipynb]: accidentally changed from full repo string replace

**Review by joshreini1 — APPROVED** [review]

**Pull request merged**